File: ProyectoAccesoDatos-AJA/src/main/java/com/example/proyecto/domain/Usuario.java
package com.example.proyecto.domain;

import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "idUsuario")
@Entity
@Table(name = "Usuario")
public class Usuario {

    public Usuario() {
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Integer idUsuario;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "apellidos")
    private String apellidos;

    @Column(name = "correo", nullable = false, unique = true)
    private String correo;

    @Column(name = "contrasena", nullable = false)
    private String contrasena;

    @Column(name = "rol", nullable = false)
    @Convert(converter = RolConverter.class)
    private Rol rol;

    // ✅ NUEVO: Columna para guardar la foto en Base64
    @Column(name = "foto_perfil", columnDefinition = "LONGTEXT")
    private String fotoPerfil;

    public enum Rol {
        ADMINISTRADOR,
        ALUMNO,
        CLIENTE
    }

    // --- GETTERS / SETTERS ---

    public Integer getIdUsuario() { return idUsuario; }
    public void setIdUsuario(Integer idUsuario) { this.idUsuario = idUsuario; }

    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }

    public String getApellidos() { return apellidos; }
    public void setApellidos(String apellidos) { this.apellidos = apellidos; }

    public String getCorreo() { return correo; }
    public void setCorreo(String correo) { this.correo = correo; }

    public String getContrasena() { return contrasena; }
    public void setContrasena(String contrasena) { this.contrasena = contrasena; }

    public void setPassword(String password) { this.contrasena = password; }

    public Rol getRol() { return rol; }
    public void setRol(Rol rol) { this.rol = rol; }

    // ✅ NUEVOS GETTERS/SETTERS
    public String getFotoPerfil() { return fotoPerfil; }
    public void setFotoPerfil(String fotoPerfil) { this.fotoPerfil = fotoPerfil; }
}

File: ProyectoFlutter/lib/models/usuario.dart
class Usuario {
  final int idUsuario;
  final String nombre;
  final String apellidos;
  final String correo;
  final String rol;

  Usuario({
    required this.idUsuario,
    required this.nombre,
    required this.apellidos,
    required this.correo,
    required this.rol,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json["idUsuario"] ?? json["id"] ?? json["id_usuario"] ?? 0,
      nombre: (json["nombre"] ?? "").toString(),
      apellidos: (json["apellidos"] ?? "").toString(),
      correo: (json["correo"] ?? "").toString(),
      rol: (json["rol"] ?? "cliente").toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "idUsuario": idUsuario,
      "nombre": nombre,
      "apellidos": apellidos,
      "correo": correo,
      "rol": rol,
    };
  }
}
